cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -228,7 +228,7 @@ private class ArrayContent extends Content, TArrayContent {
 private predicate fieldStoreStepNoChi(Node node1, FieldContent f, PostUpdateNode node2) {
   exists(StoreInstruction store, Class c |
     store = node2.asInstruction() and
-    store.getSourceValue() = node1.asInstruction() and
+    store.getSourceValueOperand() = node1.asOperand() and
```

Comment:
Won't this start picking up inexact defs? And if so, is that good or bad?

Reply:
Good point. I didn't actually think of that. For read steps we allow inexact defs (as otherwise we wouldn't pick up a load of a field address). What source code would generate a store instruction with a non-exact source value operand anyway?

Reply:
It's a register operand, so I don't think it can be inexact.